src/presenter/page-presenter.js

```diff
@@ -219,31 +249,29 @@ export default class PagePresenter {
   };
 
   #handleShowMoreButtonClick = () => {
-    const cardsCount = this.cards.length;
-    const newRenderedCardCount = Math.min(cardsCount, this.#renderedFilmsCount + FILMS_PER_CLICK);
-    const cards = this.cards.slice(this.#renderedFilmsCount, newRenderedCardCount);
+    const newRenderedCardCount = Math.min(this.#cardsList.length, this.#renderedFilmsCount + FILMS_PER_CLICK);
+    const cards = this.#cardsList.slice(this.#renderedFilmsCount, newRenderedCardCount);
 
     this.#renderFilms(cards);
     this.#renderedFilmsCount = newRenderedCardCount;
 
-    if (this.#renderedFilmsCount >= cardsCount) {
+    if (this.#renderedFilmsCount >= this.#cardsList.length) {
       remove(this.#showMoreButtonComponent);
     }
   };
 
   #findFilmsMarks = () =>
-    //при реальных данных поменять false на true
     this.cards.reduce((marks, card) => {
 
-      if(card.userDetails.watchlist === false){
+      if(card.userDetails.watchlist === true){
```

Comment:
Сам флаг `watchlist` является булевым, поэтому сравнивать на `true` нет смысла.
И два ниже аналогично.
```suggestion
      if(card.userDetails.watchlist){
```